File: fixtures/scale/scala/medium/src/calc/errors/EvalException.scala
package calc.errors

final class EvalException(message: String) extends CalcException("eval", message) {
  def withFunction(function: String): EvalException = new EvalException(function + ": " + message)
}

object EvalException {
  /** Raised when a builtin gets an argument outside its domain. */
  def domain(function: String): EvalException =
    new EvalException(function + " called outside its domain")
}

File: fixtures/scale/kotlin/medium/src/calc/errors/EvalException.kt
package calc.errors

class EvalException(message: String) : CalcException("eval", message) {

    companion object {
        /** Raised when a builtin gets an argument outside its domain. */
        fun domain(function: String): EvalException =
            EvalException(function + " called outside its domain")
    }
}
